File: contests-leetcode/weekly488/count_dominant_indices.cpp
#include<iostream>
#include<vector>

void print_arr(std::vector<int>&arr){
    for(auto it : arr){
        std::cout<<it<<" ";
    }
    std::cout<<std::endl;
}

int dominant_indices(std::vector<int>& nums) {
    int n = nums.size();
    std::vector<int>suffixArray(n,0);
    for(int i = n-2;i>=0;i--){
        suffixArray[i] = nums[i+1] + suffixArray[i+1];
    }
    
    // print_arr(suffixArray);
    
    for(int i = n-2 ;i>=0;i--){
        suffixArray[i] = suffixArray[i] / (n-i-1);
    }
    
    // print_arr(suffixArray);

    int cnt = 0;

    for(int i =0;i<n-1;i++){
        if(nums[i] > suffixArray[i]){
            cnt++;
        }
    }
    return cnt;
}

int main(){
    std::vector<int> nums = {5, 3, 2, 4, 1};
    std::cout<<dominant_indices(nums);
    return 0;
}

File: contests-leetcode/weekly488/count_dominant_indices.js
var dominantIndices = function(nums) {
    let len = nums.length;
    let helper_array = new Array(len).fill(0);
    //calculating the suffix sum array
    for(let i = len-2;i>=0;i--){
        helper_array[i] = helper_array[i+1] + nums[i+1];
    }
    //calulating the average : floor means rounding to lower value like : 3.5 to 3
    for(let i = len - 2; i>=0;i--){
        helper_array[i] = Math.floor(helper_array[i]/(len - i - 1));
    }
    let cnt = 0;

    for(let i =0;i<len-1;i++){
        if(nums[i] > helper_array[i]){
            cnt++;
        }
    }
    return cnt;
};


const arr = [5, 3, 2, 4, 1];
console.log(dominantIndices(arr));